ui/src/app/views/concept-survey-add-modal.tsx

```diff
@@ -0,0 +1,231 @@
+import {RadioButton, TextInput, ValidationError} from 'app/components/inputs';
+import {Modal, ModalBody, ModalFooter, ModalTitle} from 'app/components/modals';
+import {TooltipTrigger} from 'app/components/popups';
+import {Spinner} from 'app/components/spinners';
+import {conceptSetsApi} from 'app/services/swagger-fetch-clients';
+import {reactStyles, summarizeErrors, withCurrentWorkspace} from 'app/utils';
+import {WorkspaceData} from 'app/utils/workspace-data';
+import {
+  ConceptSet,
+  CreateConceptSetRequest,
+  Domain,
+  SurveyDetailsResponse,
+  Surveys,
+  UpdateConceptSetRequest
+} from 'generated/fetch';
+import {validate} from 'validate.js';
+
+
+const styles = reactStyles({
+  label: {
+    color: '#000',
+    paddingLeft: '0.5rem',
+    lineHeight: '19px',
+    fontSize: '14px',
+    fontWeight: 400
+  }
+});
+
+
+export const ConceptSurveyAddModal = withCurrentWorkspace()
+(class extends React.Component<{
+  workspace: WorkspaceData,
+  selectedSurvey: Array<SurveyDetailsResponse>,
+  onSave: Function,
+  onClose: Function,
+  surveyName: string
+}, {
+  conceptSets: ConceptSet[];
+  errorSaving: boolean;
+  addingToExistingSet: boolean;
+  loading: boolean;
+  nameTouched: boolean;
+  selectedSet: ConceptSet;
+  newSetDescription: string;
+  name: string;
+  saving: boolean;
+
+}> {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      conceptSets: [],
+      errorSaving: false,
+      addingToExistingSet: true,
+      loading: true,
+      nameTouched: false,
+      selectedSet: null,
+      newSetDescription: '',
+      name: '',
+      saving: false
+    };
+  }
+
+  componentDidMount() {
+    this.getExistingSurveyConceptSets();
+  }
+
+  async getExistingSurveyConceptSets() {
+    try {
+      const {workspace: {namespace, id}} = this.props;
+      const conceptSets = await conceptSetsApi().getSurveyConceptSetsInWorkspace(namespace, id,
+        this.props.surveyName);
+      const conceptSetsInSurveys = conceptSets.items
+          .filter((conceptset) => conceptset.domain === Domain.OBSERVATION);
+
+      this.setState({
+        conceptSets: conceptSetsInSurveys,
+        addingToExistingSet: conceptSetsInSurveys.length > 0,
+        loading: false,
+      });
+      if (conceptSetsInSurveys) {
+        this.setState({selectedSet: conceptSetsInSurveys[0]});
+      }
+    } catch (error) {
+      console.error(error);
+    }
+  }
+
+  async saveConcepts() {
+    const {workspace: {namespace, id}, onSave} = this.props;
+    const {selectedSet, addingToExistingSet, newSetDescription, name} = this.state;
+    this.setState({saving: true});
+    const conceptIds = this.props.selectedSurvey.map((surveys) => surveys.conceptId);
+    let survey = Surveys.THEBASICS;
+    switch (this.props.surveyName) {
+      case 'Lifestyle': survey = Surveys.LIFESTYLE; break;
+      case 'Overall Health': survey = Surveys.OVERALLHEALTH; break;
+      default: survey = Surveys.THEBASICS;
```

Comment:
Could you do `Surveys.THEBASICS` as another case, and throw an exception on default? I think that would allow us to more easily add others, and also gives us a bit more safety here that we're not seeing weird values.